arbitrum-client/src/client/mod.rs

```diff
@@ -90,42 +94,60 @@ impl ArbitrumClientConfig {
         Address::from_str(&self.darkpool_addr)
             .map_err(|e| ArbitrumClientConfigError::AddressParsing(e.to_string()))
     }
-
-    /// Constructs a [`DarkpoolContract`] instance from the configuration,
-    /// which provides strongly-typed, RPC-client-aware bindings for the
-    /// darkpool contract methods.
-    pub async fn construct_contract_instance(
-        &self,
-    ) -> Result<DarkpoolContract<MiddlewareStack>, ArbitrumClientConfigError> {
-        let rpc_client = self.get_rpc_client().await?;
-        let contract_address = self.get_darkpool_address()?;
-        let instance = DarkpoolContract::new(contract_address, rpc_client);
-        Ok(instance)
-    }
 }
 
 /// The Arbitrum client, which provides a higher-level interface to the darkpool
 /// contract for Renegade-specific access patterns.
 #[derive(Clone)]
 pub struct ArbitrumClient {
-    /// The darkpool contract instance, used to make calls to the darkpool
-    pub darkpool_contract: DarkpoolContract<MiddlewareStack>,
+    /// A list of darkpool contract clients, each configured with a different
+    /// Arbitrum account as the signer
+    darkpool_clients: Vec<DarkpoolContract<MiddlewareStack>>,
     /// The block number at which the darkpool was deployed
     deploy_block: BlockNumber,
 }
 
 impl ArbitrumClient {
     /// Constructs a new Arbitrum client from the given configuration
     pub async fn new(config: ArbitrumClientConfig) -> Result<Self, ArbitrumClientError> {
-        let darkpool_contract = config.construct_contract_instance().await?;
+        let darkpool_address = config.get_darkpool_address()?;
+        let darkpool_clients = config
+            .get_accounts()
+            .await?
+            .into_iter()
+            .map(|account| DarkpoolContract::new(darkpool_address, account))
+            .collect();
         let deploy_block = config.get_deploy_block();
 
-        Ok(Self { darkpool_contract, deploy_block })
+        Ok(Self { darkpool_clients, deploy_block })
     }
 
-    /// Get a reference to the underlying RPC client
+    /// Get a darkpool contract client
+    pub fn darkpool(&self) -> DarkpoolContract<MiddlewareStack> {
```

Comment:
nit: `get_client`, `get_signer`, or similar would be much clearer. it is unclear in the callsites what it means to call `darkpool` on the `ArbitrumClient`